pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go

```diff
@@ -33,136 +32,157 @@ func TestNewGitLabScraper(t *testing.T) {
 	assert.NotNil(t, s)
 }
 
-/*
- * Mocks
- */
-type mockClient struct {
-	BranchNames         []string
-	openMergeRequests   []getMergeRequestsProjectMergeRequestsMergeRequestConnection
-	mergedMergeRequests []getMergeRequestsProjectMergeRequestsMergeRequestConnection
-	RootRef             string
-	err                 bool
-	mergedErr           bool
-	openErr             bool
-	errString           string
-	curPage             int
+type responses struct {
+	branchResponse  branchResponse
+	mrResponse      mrResponse
+	contribResponse contribResponse
+	compareResponse compareResponse
 }
 
-func (m *mockClient) MakeRequest(ctx context.Context, req *graphql.Request, resp *graphql.Response) error {
-	switch op := req.OpName; op {
+type branchResponse struct {
+	branches     getBranchNamesProjectRepository
+	responseCode int
+}
 
-	case "getBranchNames":
-		if m.err {
-			return errors.New(m.errString)
-		}
-		r := resp.Data.(*getBranchNamesResponse)
-		r.Project.Repository.BranchNames = m.BranchNames
-		r.Project.Repository.RootRef = m.RootRef
+type mrResponse struct {
+	mrs          []getMergeRequestsProjectMergeRequestsMergeRequestConnection
+	page         int
+	responseCode int
+}
 
-	case "getMergeRequests":
-		r := resp.Data.(*getMergeRequestsResponse)
+type contribResponse struct {
+	contribs     []*gitlab.Contributor
+	responseCode int
+}
 
-		if req.Variables.(*__getMergeRequestsInput).State == "opened" {
-			if m.openErr {
-				return errors.New(m.errString)
-			}
-			if len(m.openMergeRequests) == 0 {
-				return nil
-			}
-			r.Project.MergeRequests = m.openMergeRequests[m.curPage]
-			if m.openMergeRequests[m.curPage].PageInfo.HasNextPage == false {
-				m.curPage = 0
-			} else {
-				m.curPage++
+type compareResponse struct {
+	compare      *gitlab.Compare
+	responseCode int
+}
+
+func MockServer(responses *responses) *http.ServeMux {
+	var mux http.ServeMux
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		var reqBody graphql.Request
+		if err := json.NewDecoder(r.Body).Decode(&reqBody); err != nil {
+			return
+		}
+		switch {
+		// These OpNames need to be name of the GraphQL query as defined in genqlient.graphql
+		case reqBody.OpName == "getBranchNames":
+			responses := &responses.branchResponse
```

Comment:
If I understand this right, we're creating a new scoped responses variable based on the value of the provided responses.branchResponse? Is there anyway we can use a different variable name or am I just missing something?

Reply:
updated the variable names to ones like mrResp, branchResp, etc